raiden/tests/utils/factories.py

```diff
@@ -804,37 +804,89 @@ def __init__(
 
     @property
     def channel_map(self) -> ChannelMap:
-        return {channel.identifier: channel for channel in self.channels}
+        return self.channels
+
+    @property
+    def sub_channel_map(self) -> Dict:
+        sub_channels = dict()
+        for key in self.channels.keys():
+            for sub_channel_key in self.channels[key].keys():
+                sub_channels[sub_channel_key] = self.channels[key][sub_channel_key]
+        return sub_channels
 
     @property
     def nodeaddresses_to_networkstates(self) -> NodeNetworkStateMap:
-        return {channel.partner_state.address: NODE_NETWORK_REACHABLE for channel in self.channels}
+        return {channel.partner_state.address: NODE_NETWORK_REACHABLE for channel in self.sub_channel_map.values()}
 
     def our_address(self, index: int) -> Address:
-        return self.channels[index].our_state.address
+        return self.get_sub_channel(index).our_state.address
 
     def partner_address(self, index: int) -> Address:
-        return self.channels[index].partner_state.address
+        return self.get_sub_channel(index).partner_state.address
 
-    def get_route(self, channel_index: int) -> RouteState:
-        return make_route_from_channel(self.channels[channel_index])
+    def get_route(self, key: str, channel_key: str) -> RouteState:
+        return make_route_from_channel(self.channels[key][channel_key])
 
     def get_routes(self, *args) -> List[RouteState]:
-        return [self.get_route(index) for index in (args or range(len(self.channels)))]
+        routes = []
+        for key in (args or self.channels.keys()):
+            for channel_key in self.channels[key].keys():
+                routes.append(self.get_route(key, channel_key))
+        return routes
+
+    def get_routes_by_index(self, *args) -> List[RouteState]:
+        counter = 0
+        ret_routes = []
+        for key in self.channels.keys():
+            for channel_key in self.channels[key].keys():
+                if counter in args:
+                    ret_routes.append(self.get_route(key, channel_key))
+                    counter += 1
+                else:
+                    counter += 1
+        return ret_routes
+
+    def get_route_by_index(self, *args) -> RouteState:
```

Comment:
It does not makes sense to pass *args here, args is for variable arguments, and at this case you only need to receive an index, an int. 

Also, there are ways to avoid the nested loop, you can access to for channel in self.channels.values(): to iterate over the values instead of keys.